Userland/Libraries/LibGUI/AbstractButton.cpp

```diff
@@ -32,10 +33,6 @@ AbstractButton::AbstractButton(String text)
     REGISTER_BOOL_PROPERTY("exclusive", is_exclusive, set_exclusive);
 }
 
-AbstractButton::~AbstractButton()
-{
-}
-
```

Comment:
Although using default seems ok (I'm no expert though), moving the definition to the header might mean we need more includes there. Which then transitively means more includes everywhere. Which means slower build times etc.

In this case `AbstractButton` has a `RefPtr<Core::Timer>` because of which we don't need to include it in the header. But it is needed to call the destructor of `RefPtr<Core::Timer>`.

(The build failed because of unfortunate timing it seems, so I can't say conclusively)
And even if this builds maybe it is more worth it to prune the includes.

Reply:
Agreed. Moving to the header has several effects. It *can* mean more includes. I also results in more optimizations being available. Especially for classes which have trivial constructors/destructors. Putting them into a different translation unit inhibits inlining without using LTO, afaik. Additionally, it means more work the linker has to do to resolve symbols. It is definitely a balance that needs to be played.

Reply:
I just ran multiple clean builds of LibGUI on my machine setting the CPU speed so that scaling is disabled and removing ccache artifacts. With my changes I repeatedly get about 0.5 seconds faster for the ~102 seconds it takes. This is not at all scientific, but leads me to believe there is not a significant difference here.